x-pack/plugins/ml/public/application/components/data_grid/feature_importance/use_classification_path_data.tsx

```diff
@@ -139,22 +143,127 @@ export const buildRegressionDecisionPathData = ({
     (f) => f !== undefined
   ) as ExtendedFeatureImportance[];
 
-  return buildDecisionPathData(filteredFeatureImportance);
+  const finalResult: DecisionPathPlotData = filteredFeatureImportance
+    // sort by absolute importance so it goes from bottom (baseline) to top
+    .sort(
+      (a: ExtendedFeatureImportance, b: ExtendedFeatureImportance) =>
+        b.absImportance - a.absImportance
+    )
+    .map((d) => [d[FEATURE_NAME] as string, d[FEATURE_IMPORTANCE] as number, NaN]);
+
+  // start at the baseline and end at predicted value
+  // for regression, cumulativeSum should add up to baseline
+  let cumulativeSum = 0;
+  for (let i = filteredFeatureImportance.length - 1; i >= 0; i--) {
+    cumulativeSum += finalResult[i][1];
+    finalResult[i][2] = cumulativeSum;
+  }
+  return finalResult;
+};
+
+export const buildBinaryClassificationDecisionPathData = ({
```

Comment:
this function actually mutates `decisionPlotData`, hence there is no need to return it. 
I'd suggest doing the following: 
1. call the function `processBinaryClassificationDecisionPathData ` and keep it as is.
or 
2. call the function `getBinaryClassificationDecisionPathData` and implement it as a pure function 

the same is applicable to `buildMultiClassClassificationDecisionPathData`

Reply:
That's a good point. Renamed the functions here https://github.com/elastic/kibana/pull/80229/commits/75261270409ee829d65ab9c989c4c8c60ac67656.